EcoTrackerApp/EcoTrackerApp/SpeciesPage/SpeciesPageVC.swift

```diff
@@ -5,24 +5,278 @@
+            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -10)
+        ])
+    }
+    
+    private func setupSearchBar() {
+        searchBar.tintColor = .white
+        
+        let textField = searchBar.value(forKey: "searchField") as? UITextField
+        textField?.attributedPlaceholder = NSAttributedString(string: "Enter a city name", attributes: [NSAttributedString.Key.foregroundColor: UIColor.systemCyan])
+        
+        navigationItem.titleView = searchBar
+    }
+}
 
+// MARK: - UITableViewDataSource Extension
+extension SpeciesPageVC: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        viewModel.numberOfSpecies()
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: SpeciesPageTableViewCell.reuseIdentifier, for: indexPath) as? SpeciesPageTableViewCell else {
+            fatalError("Unable to dequeue SpeciesPageTableViewCell")
+        }
+        
+        let species = viewModel.species(at: indexPath.row)
+        cell.configure(with: species)
+        
+        return cell
     }
     
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let species = viewModel.species(at: indexPath.row),
+              let wikipediaURLString = species.speciesWikipediaLink,
+              let wikipediaURL = URL(string: wikipediaURLString) else {
+            return
+        }
+        UIApplication.shared.open(wikipediaURL, options: [:], completionHandler: nil)
+    }
+}
+
+// MARK: - UITableViewDelegate Extension
 
-    /*
-    // MARK: - Navigation
+extension SpeciesPageVC: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+            200.0
+        }
+}
 
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+// MARK: - UISearchBarDelegate Extension
+extension SpeciesPageVC: UISearchBarDelegate {
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let cityName = searchBar.text, !cityName.isEmpty else {
+            print("City name is empty")
+            viewModel.clearSpeciesData()
+            tableView.reloadData()
+            return
+        }
+        
+        print("Fetching data for city: \(cityName)")
+        
+        viewModel.fetchData(for: cityName) { [weak self] result in
```

Comment:
კონტროლერში სერვის ქოლის რესფონსის success and failure არუნდაიჰენდლებოდეს. ეს მოდელის საქმეა